Feat: add HTML link support for index page featured description; Fix: title, description and links colors

beamariano opened the pull request:
## Description
This feature allows for the description field of a playlist featured on the home page to render links and other font stylings. This is implemented by updating the model, creating validators (only specific html tags allowed), and adding dangerouslySetInnerHTML to the div that renders the description. It also fixes the title and description text not being shown due to the color being white. Link colors are also updated on dark mode.

## Related Issue
Fixes #(issue) https://github.com/EngageMedia-video/cinematacms/issues/192 and https://github.com/EngageMedia-video/cinematacms/issues/191

## Motivation and Context
Presently the description field renders only plain text. This feature allows the description to contain internal links, bold, italics, paragraph breaks and a new paragraph.

## How Has This Been Tested?
Graybox

## Screenshots (if appropriate):
<img width="545" height="468" alt="Screenshot 2025-10-08 at 10 14 10 PM" src="https://github.com/user-attachments/assets/0247fd31-2037-4613-be87-94a2c62ce41d" />

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue) [Text and link colors]
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes. [Test was written by Claude]
- [x] All new and existing tests passed.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Rich-text descriptions supported for media/category sections with safe internal link handling.
  - Improved admin editing for featured content with HTML validation/sanitization and better form integration.
  - Enhanced forms integration and templates (improved form rendering and optional captcha support).

- Bug Fixes
  - HTML sanitization blocks unsafe tags/attributes and external or dangerous links.

- Style
  - Refreshed homepage category styling with full light/dark theme support.

- Tests
  - Comprehensive tests for URL validation and HTML sanitization.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

beamariano committed:
feat: add models, validators, etc files

beamariano committed:
feat: add models, validators, etc files

beamariano committed:
feat: add sanitize html and fix installed apps

beamariano committed:
feat: reject if tags other than those allowed are submitted

beamariano committed:
feat: add dangerouslySetInnerHTML in description component

beamariano committed:
Merge branch 'main' into html-link-support-for-index-featured-description

adryanev committed:
fix: update Dependabot configuration to use weekly schedule for all p… (#277)

## Description
Updated Dependabot configuration in
[.github/dependabot.yml](.github/dependabot.yml) to optimize automated
dependency management:

- **Changed update frequency** from daily to weekly for all package
ecosystems (pip, uv, and npm)
- **Added dependency grouping** to consolidate related updates into
single PRs:
  - `python-dependencies` group for pip and uv packages
  - `frontend-dependencies` group for npm packages
- **Added ignore rules** for React major version updates to prevent
breaking changes
- **Reordered package ecosystems** (pip before uv) for better
organization

## Related Issue
<!--- Please link to the issue here: -->
Fixes #(issue)

## Motivation and Context
This change addresses several pain points with the current Dependabot
configuration:

1. **Reduce PR noise**: Daily updates generate excessive PRs that can
overwhelm the team. Weekly updates provide a better balance between
staying current and maintainability.
2. **Consolidated updates**: Grouping dependencies means fewer PRs to
review, as related packages are updated together in a single PR.
3. **Prevent breaking changes**: Ignoring React major version updates
protects against unintentional breaking changes that require significant
refactoring.
4. **Better workflow integration**: Works seamlessly with the existing
[dependabot-check.yml](.github/workflows/dependabot-check.yml) workflow
that validates dependency resolution.

## How Has This Been Tested?
- Validated YAML syntax in
[.github/dependabot.yml](.github/dependabot.yml)
- Verified configuration follows [Dependabot v2 schema
specifications](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file)
- Confirmed grouping patterns and ignore rules are correctly formatted
- Checked that the configuration is compatible with existing CI
workflows

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to change)
- [x] Documentation update

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.


<!-- This is an auto-generated comment: release notes by coderabbit.ai
-->

## Summary by CodeRabbit

- Chores
- Standardized dependency update schedules to weekly for Python and
JavaScript projects to reduce PR noise.
- Grouped Python and frontend dependency updates into consolidated PRs
for easier review.
- Switched Python ecosystem configuration to pip for consistency across
tooling.
- Added ignore rules to skip semver-major updates for React and React
DOM, minimizing disruptive changes.
- No user-facing changes; this improves maintenance workflow and
dependency management cadence.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

adryanev committed:
Docs: makefile installation (#276)

## Description
This PR improves the Makefile commands and updates the documentation for
better clarity and usability:

1. **Makefile improvements:**
- Added `--detach` option to all Celery start commands (beat, long,
short, whisper) to run workers in background mode
- Renamed `build-all` command to `quick-build` for better clarity about
what it does (main app only)
   - Updated help text to reflect the new `quick-build` command name

2. **Documentation updates:**
- Fixed grammar in Celery worker commands documentation (singular/plural
consistency)
- Added new "Frontend Commands" section documenting all available
frontend-related make commands
- Improved documentation structure in
[docs/setup/makefile-and-uv.md](docs/setup/makefile-and-uv.md)

## Related Issue
<!--- Please link to the issue here: -->
Fixes #(issue)

## Motivation and Context
The previous Celery start commands were running in foreground mode,
blocking the terminal. Adding the `--detach` option allows workers to
run in the background, improving the developer experience.

The documentation was incomplete, missing details about frontend
commands, and had some grammatical inconsistencies that needed
correction.

## How Has This Been Tested?
- Verified Makefile commands execute without syntax errors
- Confirmed `--detach` option is valid for Celery worker commands
- Reviewed documentation for accuracy and completeness

## Screenshots (if appropriate):
N/A

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all
the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to change)
- [x] Documentation update

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes
that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're
here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.


<!-- This is an auto-generated comment: release notes by coderabbit.ai
-->
## Summary by CodeRabbit

- New Features
  - Added a quick-build command for faster builds.
- Celery start commands now support background execution (detached) for
multiple worker types.
  - Replaced the previous build-all command with quick-build.

- Documentation
- Expanded Makefile docs with a Frontend Commands section and clarified
Celery command usage.
  - Updated wording (worker → workers) and tables for clarity.
- Improved Mac setup guides: use mkdir -p, added media_files/hls
directory, and clarified mp4hls binary location.

- Chores
  - Removed the deprecated build-all command.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

beamariano committed:
feat: add missing dependency

beamariano committed:
fix: render description

beamariano committed:
revert: commented out language

adryanev commented on frontend/src/static/js/pages/components/MediaListHeader.js:
```diff
@@ -6,7 +6,7 @@ export function MediaListHeader(props){
 				{ void 0 === props.viewAllLink || null === props.viewAllLink ? null : <a href={ props.viewAllLink } title={ props.viewAllText || props.viewAllLink } className="hw-category-link">{ props.viewAllText || props.viewAllLink }</a> }
 			</div>
-			{ props.desc && <div className="hw-category-description">{props.desc}</div> }
+			{ props.desc && <div className="hw-category-description" dangerouslySetInnerHTML={{__html: props.desc}}/> }
```
agree with coderabbit

dependabot[bot] committed:
chore(deps): bump the frontend-dependencies group in /frontend with 2 updates (#281)

chore(deps): bump the frontend-dependencies group

Bumps the frontend-dependencies group in /frontend with 2 updates: [axios](https://github.com/axios/axios) and [dotenv](https://github.com/motdotla/dotenv).


Updates `axios` from 1.12.1 to 1.12.2
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.12.1...v1.12.2)

Updates `dotenv` from 17.2.2 to 17.2.3
- [Changelog](https://github.com/motdotla/dotenv/blob/master/CHANGELOG.md)
- [Commits](https://github.com/motdotla/dotenv/compare/v17.2.2...v17.2.3)

---
updated-dependencies:
- dependency-name: axios
  dependency-version: 1.12.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: frontend-dependencies
- dependency-name: dotenv
  dependency-version: 17.2.3
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: frontend-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

adryanev committed:
fix: media languange, country, celery task (#287)

* Refactor code for improved readability and maintainability

- Organized imports in models.py for better clarity.
- Enhanced exception handling in transcription tasks to ensure proper cleanup on failure.
- Updated logging messages for better context and error tracking.
- Refactored media and language handling in update_listings_thumbnails to utilize database queries directly.
- Improved command execution in tasks.py to avoid shell injection vulnerabilities.
- Cleaned up URL patterns in urls.py to remove unnecessary comments and improve structure.
- Standardized string formatting and spacing across various files for consistency.
- Added detailed comments and docstrings to clarify functionality and intent.

* fix: simplify country retrieval logic in media_save function

* feat: add command to populate Topic table from video_topics list

* feat: add commands to populate media languages, countries, and topics in setup scripts

* docs: update setup instructions to make build scripts executable

* feat: filter topics to include only those with associated media

* feat: enhance topic population command with case-insensitive matching and improved dry-run output

* refactor: remove redundant logging imports and unused Lower function

* chore: remove duplicated logger assignment

adryanev committed:
fix: run the cron to every 30 minutes (#291)

* fix: run the cron to every 30 minutes

* fix: standardize string quotes and improve comments in settings.py

beamariano committed:
feat: add more robust url validation

beamariano committed:
feat: add sanitize html and validation improvements; add unit tests

beamariano committed:
feat: add validator - allow from internal domain

beamariano committed:
feat: add beautiful soup; remove unusued code

beamariano committed:
chore: remove duplicate imports

beamariano committed:
chore: remove duplicate code

beamariano committed:
feat: persist sanitized; chore: refactor tags and attribs constants

beamariano committed:
feat: sanitize at save so that unsanitized html is not stored

beamariano committed:
fix: sanitize and save at correct part

beamariano committed:
fix: link colors and text colors for light and dark mode as well as odd/even sections

beamariano committed:
Merge branch 'main' into featured-index-html

beamariano committed:
feat: add beautifulsoup to requirements.txt

beamariano committed:
fix: add default empty string to text field

beamariano committed:
refactor: remove beautiful soup, use regex, add missing from new specifications

adryanev commented on files/models.py:
```diff
@@ -1739,6 +1742,18 @@ class Meta:
+    class Meta:
+        ordering = ["ordering"]
+        verbose_name = "Index page featured"
+        verbose_name_plural = "Index page featured"
```
please fix this @beamariano

beamariano committed:
Update files/models.py

remove redundant Meta

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
